tools/LogCollector/source/omiHighCPUDiagnostics.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+#set -x
+declare -a ppuc
+declare -a pt
+count=2880
+sleep_time_btw_each_collection=5 #seconds
+
+if [ "$1" = "--help" ]; then
+    echo "This scipt runs for 24 hour and collect info about omiagent process with high cpu utilization(>20%)."
+    echo "When CPU utilization is high it takes the stack trace snapshot for the omiagent for 10 times in interval of 10 sec."
+    echo "Then rescan the system in interval of 10sec."
+    echo "It also collects the diff of scx and omi logs in every 10 sec"
+    echo "  Option            Description"
+    echo "--runtime-in-min    set run time in minutes, default 3600 -> 24 hours"
+    echo "--trace-enable      set verbose logging for omi and scx with option trace-enable"
+    echo "--trace-disable     disable verbose logging for omi and scx"
+    exit 0;
+
+elif [ "$1" = "--trace-enable" ]; then
+    /opt/microsoft/scx/bin/tools/scxadmin -log-set all intermediate
+    sed -i 's/#*loglevel/loglevel/' /etc/opt/omi/conf/omiserver.conf
+    sed -i 's/WARNING/DEBUG/' /etc/opt/omi/conf/omiserver.conf
+    /opt/microsoft/scx/bin/tools/scxadmin -restart
+    exit 0
+
+elif [ "$1" = "--trace-disable" ]; then
+    /opt/microsoft/scx/bin/tools/scxadmin -log-set all errors
+    sed -i 's/loglevel/#loglevel/' /etc/opt/omi/conf/omiserver.conf
+    /opt/microsoft/scx/bin/tools/scxadmin -restart
+    exit 0
+elif [ "$1" = "--runtime-in-min" ]; then
+    shift
+    if [ $1 -gt 0 ]; then
+        let "count = $1*60/$sleep_time_btw_each_collection"
+    else
+        echo "runtime value is invalid '$1', should be greater than or equal 1" >& 2; exit 1;
```

Comment:
runtime value **of '$1'** is invalid, should be greater than or equal **to** 1

Reply:
done